Fails to build on 32 bit arch: incompatible pointer type

### Actual behavior
when we build the latest versions of openvas-scanner in Debian, it fails to build on i386 and armhf.

```
/build/reproducible-path/openvas-scanner-23.40.2/nasl/nasl_misc_funcs.c: In function ‘nasl_gunzip’:
/build/reproducible-path/openvas-scanner-23.40.2/nasl/nasl_misc_funcs.c:830:49: error: passing argument 3 of ‘gvm_uncompress’ from incompatible pointer type [-Wincompatible-pointer-types]
  830 |   uncompressed = gvm_uncompress (data, datalen, &uncomplen);
      |                                                 ^~~~~~~~~~
      |                                                 |
      |                                                 size_t * {aka unsigned int *}
In file included from /build/reproducible-path/openvas-scanner-23.40.2/nasl/nasl_misc_funcs.c:31:
/usr/include/gvm/util/compressutils.h:23:46: note: expected ‘long unsigned int *’ but argument is of type ‘size_t *’ {aka ‘unsigned int *’}
   23 | gvm_uncompress (const void *, unsigned long, unsigned long *);
      |                                              ^~~~~~~~~~~~~~~
/build/reproducible-path/openvas-scanner-23.40.2/nasl/nasl_misc_funcs.c: In function ‘nasl_gzip’:
/build/reproducible-path/openvas-scanner-23.40.2/nasl/nasl_misc_funcs.c:858:60: error: passing argument 3 of ‘gvm_compress_gzipheader’ from incompatible pointer type [-Wincompatible-pointer-types]
  858 |       compressed = gvm_compress_gzipheader (data, datalen, &complen);
      |                                                            ^~~~~~~~
      |                                                            |
      |                                                            size_t * {aka unsigned int *}
/usr/include/gvm/util/compressutils.h:20:55: note: expected ‘long unsigned int *’ but argument is of type ‘size_t *’ {aka ‘unsigned int *’}
   20 | gvm_compress_gzipheader (const void *, unsigned long, unsigned long *);
      |                                                       ^~~~~~~~~~~~~~~
/build/reproducible-path/openvas-scanner-23.40.2/nasl/nasl_misc_funcs.c:864:49: error: passing argument 3 of ‘gvm_compress’ from incompatible pointer type [-Wincompatible-pointer-types]
  864 |       compressed = gvm_compress (data, datalen, &complen);
      |                                                 ^~~~~~~~
      |                                                 |
      |                                                 size_t * {aka unsigned int *}
/usr/include/gvm/util/compressutils.h:17:44: note: expected ‘long unsigned int *’ but argument is of type ‘size_t *’ {aka ‘unsigned int *’}
   17 | gvm_compress (const void *, unsigned long, unsigned long *);
      |                                            ^~~~~~~~~~~~~~~
make[4]: *** [nasl/CMakeFiles/openvas_nasl_shared.dir/build.make:540: nasl/CMakeFiles/openvas_nasl_shared.dir/nasl_misc_funcs.c.o] Error 1
```

I think it has been introduced by the commit  https://github.com/greenbone/openvas-scanner/pull/2110/changes/13c2f987ad7b49ab8a6260cf7ed350636ad10a9a
and the type changes in ‎nasl/nasl_misc_funcs.c

### GVM versions

openvas: latest version 23.40.2 fails to build (version 23.38.5 fails too)

gvm-libs: 22.35.8

### Environment

Operating system: Debian Unstable

--- a/nasl/nasl_misc_funcs.c
+++ b/nasl/nasl_misc_funcs.c
@@ -817,7 +817,7 @@ nasl_gunzip (lex_ctxt *lexic)
 {
   tree_cell *retc;
   void *data, *uncompressed;
-  size_t datalen, uncomplen;
+  unsigned long datalen, uncomplen;
 
   data = get_str_var_by_name (lexic, "data");
   if (data == NULL)
@@ -842,7 +842,7 @@ nasl_gzip (lex_ctxt *lexic)
 {
   tree_cell *retc;
   void *data, *compressed, *headerformat;
-  size_t datalen, complen;
+  unsigned long datalen, complen;
 
   data = get_str_var_by_name (lexic, "data");
   if (data == NULL)

